inc/ptw.h

```diff
@@ -0,0 +1,142 @@
+#ifndef PTW_H
+#define PTW_H
+
+#include "cache.h"
+#include <map>
+
+#define IS_PSCL5 11
+#define IS_PSCL4 12
+#define IS_PSCL3 13
+#define IS_PSCL2 14
+
+#define PSCL5_SET 1
+#define PSCL5_WAY 2
+
+#define PSCL4_SET 1
+#define PSCL4_WAY 4
+
+#define PSCL3_SET 2
+#define PSCL3_WAY 4
+
+#define PSCL2_SET 4
+#define PSCL2_WAY 8
+
+#define PTW_RQ_SIZE STLB_MSHR_SIZE
+#define PTW_WQ_SIZE 0
+#define PTW_MSHR_SIZE 1
```

Comment:
Only one inflight PTW seems unnecessarily restrictive. Given the potentially long latency of the walks, it's realistic that a few may be happening in parallel.

Reply:
I think that building the inner workings of the page miss handler on top of the MSHR isn't quite the right thing to do at all.  I think that other than the producer/consumer interfaces between the STLB on one side and the caches on the other, the page table walker merits being its own thing that isn't just cobbled together from recycled parts.

Having said that, I think it's fine how it is for a first draft.  I wouldn't require this to be changed for the initial PR, and I think we could iterate on it in the long term.

Reply:
I'm not sure if I understand you. I think "MSHR" here is just a term to mean "in-progress page walks". It's not an MSHR in the same sense as the caches have.

Reply:
I guess I agree that "miss status handling register" is not a bad name for this item.  Anyway, I still think this is something that could wait to be changed later.